File: 104/wordToDigit.java
/*
	@author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
	Description of problem at: https://www.codeeval.com/open_challenges/104/
*/

import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

public class wordToDigit {
	public static void main(String[] args) {
		try {
			File file = new File(args[0]);
			BufferedReader infile = new BufferedReader(new FileReader(file));
			String line;
			String[] numbers = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
	
			while ((line = infile.readLine()) != null) {
				String[] words = line.split(";");
				for (int i = 0; i < words.length; i++) {
					for (int j = 0; j < numbers.length; j++) {
						if (words[i].compareTo(numbers[j]) == 0) {
							System.out.print(j);
							break;
						}
					}
				}
				System.out.println();
			}
			infile.close();
	
		} catch (FileNotFoundException e) {
			System.out.println("This is crap. Wrong stupid file name.");
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}

File: 104/wordtodigit.js
/*
    @author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
    Description of problem at: https://www.codeeval.com/open_challenges/104/
*/

var fs = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line !== "") {
        line = line.replace(/\r/gm, ""); // For Windows.
        var arr = line.split(';');
		var x;
		var digits = [""];
		for (x in arr) {
			switch(arr[x]) {
				case "zero":
					digits[x] = "0";
					break;
				case "one":
					digits[x] = "1";
					break;
				case "two":
					digits[x] = "2";
					break;
				case "three":
					digits[x] = "3";
					break;
				case "four":
					digits[x] = "4";
					break;
				case "five":
					digits[x] = "5";
					break;
				case "six":
					digits[x] = "6";
					break;
				case "seven":
					digits[x] = "7";
					break;
				case "eight":
					digits[x] = "8";
					break;
				case "nine":
					digits[x] = "9";
					break;
			}
		}
		for (x in digits) {
			process.stdout.write(digits[x]);
		}
		console.log();
    }
});
